ign -> gz Migrate Ignition headers

nkoenig opened the pull request:
See: https://github.com/gazebo-tooling/release-tools/issues/784

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

scpeters commented on include/ignition/utils/config.hh:
```diff
@@ -0,0 +1,39 @@
+*/
+
+#ifndef IGNITION_GUI__CONFIG_HH_
+#define IGNITION_GUI__CONFIG_HH_
```
replace `GUI` with `UTILS`

nkoenig replied:
[e1c99ab](https://github.com/gazebosim/gz-utils/pull/84/commits/e1c99ab5d67d2c6b04ca43c1b117399892ba8489)

scpeters reviewed (commented):
do you mind squashing the last 3 commits? It's easier to review if the first commit just renames the headers and then a second commit modifies things

nkoenig commented:
> do you mind squashing the last 3 commits? It's easier to review if the first commit just renames the headers and then a second commit modifies things

Done.

scpeters commented on include/ignition/utils/config.hh:
```diff
@@ -0,0 +1,39 @@
+ *
+*/
+
+#ifndef GZ_GUI__CONFIG_HH_
+#define GZ_GUI__CONFIG_HH_
```
```suggestion
#ifndef IGNITION_UTILS__CONFIG_HH_
#define IGNITION_UTILS__CONFIG_HH_
```

nkoenig replied:
Done.

scpeters commented on include/gz/utils/NeverDestroyed.hh:
```diff
@@ -0,0 +1,162 @@
+/// Foo ParseFoo(const std::string& foo_string)
+/// {
+///   using Dict = std::unordered_map<std::string, Foo>;
+///   static const igntion::utils::NeverDestroyed<Dict> string_to_enum
```
```suggestion
///   static const ignition::utils::NeverDestroyed<Dict> string_to_enum
```

nkoenig replied:
Done

nkoenig commented:
Should I also update the contents of the `cli` directory?

scpeters commented:
> Should I also update the contents of the `cli` directory?

yes, good point. I would follow the pattern on the [`gz-utils2` branch](https://github.com/gazebosim/gz-utils/tree/gz-utils2/cli/include)

nkoenig commented:
> > Should I also update the contents of the `cli` directory?
> 
> yes, good point. I would follow the pattern on the [`gz-utils2` branch](https://github.com/gazebosim/gz-utils/tree/gz-utils2/cli/include)

Great. I update the CLI directory in a few commits that should make it easy to review:

1. Moved ignition directories to gz: [01326df](https://github.com/gazebosim/gz-utils/pull/84/commits/01326dfbb099e34f56c024b1e946b3f32e4082e9)
2. Copied the ignition directories from gz-utils2: [9822a1a](https://github.com/gazebosim/gz-utils/pull/84/commits/9822a1a488f22caa28eafe6d684df3f1d525b984)
3. Renamed IgnitionFormatter.hpp: [b539c89](https://github.com/gazebosim/gz-utils/pull/84/commits/b539c896ddc26a3dba13a6d673632a25016b7630)
4. Updated CMake and gz naming in GzFormatter.hpp: [91e1cda](https://github.com/gazebosim/gz-utils/pull/84/commits/91e1cdac1b0d26af5ee76d054170ae65d9f14173)

scpeters reviewed (changes requested):
I have some small changes I'd like to make to whitespace and squashing commits and will push directly to this branch with my fixes later today

scpeters reviewed (approved):
I'm made some more changes and squashed putting all file renames in the first commit, most documentation changes in the last commit, and everything else in the middle commit
